cpp/include/raft/cuda_utils.cuh

```diff
@@ -694,6 +694,41 @@ DI T blockReduce(T val, char* smem)
   return warpReduce(val);
 }
 
+/**
+ * @defgroup Load/Store Functions with cache control (for clang compilation)
+ * @{
+ */
+template <typename T>
+__device__ __forceinline__ T ldcg(const T* addr)
+{
+#if defined(__clang__)
+  return *addr;
+#else
+  return __ldcg(addr);
+#endif
+}
+
+template <typename T>
+__device__ __forceinline__ T ldcv(const T* addr)
+{
+#if defined(__clang__)
+  return *addr;
+#else
+  return __ldcv(addr);
+#endif
+}
+
+template <typename T>
+__device__ __forceinline__ void stwt(T* addr, const T& val)
+{
+#if defined(__clang__)
+  *addr = val;
```

Comment:
Maybe we'd need some `volatile`-casting tricks here to make clang backend avoid caches as well?